### PR TITLE
feat: allow disabling tokenizers

### PR DESCRIPTION
**Pull request by kellyjosephprice**
[][demo] | RM-215
:---:|:---:

## 🧰  Changes

Adds the option of disabling tokenizers. This is in service of the editor's markdown decorators. During deserialization, we can leave the inlines as plain text. Then, during the decoration step, we can parse the markdown with just the inline tokenizers. 

## Usage
```
  const opts = { tokenizerSet: 'blocks' };
  markdown.mdast(md, opts); // won't parse inline code, emphasis, etc.
```

[demo]: https://markdown-pr-119.herokuapp.com

### CODE REVIEW

**Comment by rafegoldberg** [comment]
Can you add a brief usage example to the PR description?

File: __tests__/disabling-tokenizers.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const markdown = require('../index');
+
+test('disabling inlineCode', () => {
+  const md = '`const js = true `';
+  const opts = { disableTokenizers: { inline: ['code'] } };
```
Comment (rafegoldberg) [inline __tests__/disabling-tokenizers.test.js:5]: Ahhh, I see. I'd prefer if we could just pass a simple boolean in to the option. Does the user really need such fine grained control over what is and isn't considered inline? Can we make that decision for them?

Comment (kellyjosephprice) [reply __tests__/disabling-tokenizers.test.js:5]: Well, the driving force is when the editor deserializes markdown, it doesn't parse inlines and gets back plain text. Then it parses that plain text with just the inlines enabled to do the decoration.

Right now, my plan is to disable all inlines except `links` during deserialization. From a development standpoint, I'd like to be able to twiddle the tokenizers while I figure things out.

Comment (kellyjosephprice) [reply __tests__/disabling-tokenizers.test.js:5]: As per slack, I'll update this with simpler options.

**Review by gratcliff — APPROVED** [review]

File: options.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    paddedTable: true,
+    setext: true,
+  },
+  normalize: true,
+  settings: {
+    position: false,
+  },
+};
+
+// NOTE: disabling newline, paragraph, or text trips remark into an infinite loop!
```
Comment (rafegoldberg) [inline options.js:21]: I can haz JSDOC comments?

```suggestion
/**
 * @note disabling `newline`, `paragraph`, or `text` tokenizers trips remark into an infinite loop!
 */
```

File: options.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    },
+  },
+};
+
+const parseOptions = (opts = {}) => {
+  if (opts.tokenizerSet in disableTokenizers) {
+    return { ...disableTokenizers[opts.tokenizerSet], ...opts };
+  } else if (opts.tokenizerSet) {
+    throw new Error(`opts.tokenizerSet "${opts.tokenizerSet}" not one of "${Object.keys(disableTokenizers)}"`);
+  }
```
Comment (rafegoldberg) [inline options.js:80]: Can we rename the public-facing option to `disableTokenizers`? It feels a bit redundant here, but I think it'd be a bit more descriptive of what it actually does for the end user.
```suggestion
  if (opts.disableTokenizers in disableTokenizers) {
    return { ...disableTokenizers[opts.disableTokenizers], ...opts };
  } else if (opts.disableTokenizers) {
    throw new Error(`opts.disableTokenizers "${opts.disableTokenizers}" not one of "${Object.keys(disableTokenizers)}"`);
  }
```

File: __tests__/link-parsers.test.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+test('a bracketed autoLinked url', () => {
+  expect(markdown.mdast('<http://www.google.com>')).toMatchSnapshot();
+});
+
+test('a bare autoLinked url', () => {
+  expect(markdown.mdast('http://www.googl.com')).toMatchSnapshot();
+});
+
+test('a bare autoLinked url with no protocol', () => {
```
Comment (rafegoldberg) [inline __tests__/link-parsers.test.js:19]:
```suggestion
test.todo('a bare autoLinked url with no protocol', () => {
```

**Review by rafegoldberg — APPROVED** [review]
I had some suggestions, but nothing I'd hold merging this over!

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.26.0-next.1</b>](/readmeio/markdown/releases/tag/v6.26.0-next.1)
<br>

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.26.0</b>](/readmeio/markdown/releases/tag/v6.26.0)
<br>

**Pull request merged**